### PR TITLE
Ui.msoper -- bringing in the UI with rebase and squash

### PR DESCRIPTION
**Pull request by marksoper**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,11 +3,11 @@
 Docker Volume Driver for vSphere
 ================================
 
-This repo hosts the Docker Volume Driver for vSphere. Docker Volume Driver enables customers to address persistent storage requirements for Docker containers in vSphere environments. This plugin is integrated with [Docker Volume Plugin framework](https://docs.docker.com/engine/extend/plugins_volume/). Docker users can now consume vSphere Storage (vSAN, VMFS, NFS) to address persistency requirements of containerized cloud native apps using Docker Ecosystem. 
+This repo hosts the Docker Volume Driver for vSphere. Docker Volume Driver enables customers to address persistent storage requirements for Docker containers in vSphere environments. This plugin is integrated with [Docker Volume Plugin framework](https://docs.docker.com/engine/extend/plugins_volume/). Docker users can now consume vSphere Storage (vSAN, VMFS, NFS) to address persistency requirements of containerized cloud native apps using Docker Ecosystem.
```
Comment (pdhamdhere) [inline README.md]: Are there any changes in files? There are whitespace changes in this & other files. Can you please remove those. Makes reviewing really hard. If there is no change, please remove this file from PR.

File: ui/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# EditorConfig helps developers define and maintain consistent
```
Comment (pdhamdhere) [inline ui/.editorconfig]: Do we want to merge this file?

Comment (marksoper) [inline ui/.editorconfig]: yes, it contains meaningful information - such as the indent increment size (2) and style (soft) - to help a other developers be consistent

**Comment by marksoper** [comment]
I've looked at this for cleanup and don't see that any further changes are necessary. Here's what I reviewed and why I don't think change is needed in each category.

I'm very open to different opinions, please LMK what you think.

**1) New updates to Master**

some of the diffs here are result of updates to master since we rebased and submitted this PR. So those changes show up as diffs here (for example README.md changes are from (asomaiah/patch-1) https://github.com/vmware/docker-volume-vsphere/commit/90c661b01a03dbc27d0f06f26a9ddb8d48f16736

We could rebase on the newer master and resubmit the PR, but this rebase process was complicated, involving lots of manual conflict resolution.

RECOMMENDATION: DO NOTHING

**2) All the files in /ui**

This looks good, all files are required files. 

RECOMMENDATION: DO NOTHING

**3) Changes made by Ritesh and Msoper outside of /ui**

There several such files: e.g. Makefile, esx_service/Makefile, etc.
These changes are required. 

RECOMMENDATION: DO NOTHING

**4) Configuration files**

Config files like .gitignore, .editorconfig, .eslintrc
I included them in version control because they contain valuable information to help other developers stay consistent in their use of Javascript (which doesn't otherwise have strong style guidance) and help indicate which dirs (e.g. /build) are artifacts.

There's a reasonable argument for both leaving them in and taking them out. It's easier to leave them in. We can remove them at any time.

RECOMMENDATION: DO NOTHING

**5) Whitespace changes **

Whitespace changes seem to occur only in files that have other changes, like adding extra lines between statements as matter of coding preference. That is bound to happen as code evolves. I don't see any big issues like massive whitespace changes from editor tab style inconsistency, etc.

RECOMMENDATION: DO NOTHING

File: README.md [hunk 3]
```diff
@@ -1,9 +1,11 @@
 [![Build Status](https://ci.vmware.run/api/badges/vmware/docker-volume-vsphere/status.svg)](https://ci.vmware.run/vmware/docker-volume-vsphere)
 
-Docker Volume Driver for vSphere
```
Comment (pdhamdhere) [inline README.md]: Please rebase. There should be no conflicts if you haven;t touched the file.  `git fetch; git pull --rebase master` should do the trick. Please check with @kerneltime if you need help.

File: Makefile [hunk 4]
```diff
@@ -16,17 +16,36 @@
...
+	clean-all clean-vm clean-esx \
+	build-all dockerbuild \
 	deb rpm package gvt documentation
 
-# default target
-default: build-all
+BUILD := ./misc/scripts/build.sh
+
+# default target, build ui then build vib, rpm, deb
+default: dockerized-build-ui build-all
```
Comment (pdhamdhere) [inline Makefile]: We should not include UI build in default make and should not include UI files in VIB until feature is complete and ready to be released. Until then, please have separate build target for UI and vib.

File: README.md (hunk 3 above)
Comment (pdhamdhere) [inline README.md]: PLease rebase.

File: esx_service/Makefile [hunk 5]
```diff
@@ -113,11 +120,12 @@ SO_SQLITE := tools/sqlite/*.so
 # copy files to staging area and then build the VIB
 $(VIB_BIN): $(TO_WDIR) $(TO_ESX_BIN) $(TO_ESX_PY) $(TO_ESX_LIB) $(TO_ESX_INITD) $(PY_SQLITE) $(SO_SQLITE)
 	@echo Staging in $(PAYLOAD)
-	@mkdir -p $(VMDKOPS_BIN) $(VMDKOPS_PY) $(VMDKOPS_PY2) $(VMDKOPS_LIB) $(VMDKOPS_LIB64) $(VMDKOPS_INITD)
+	@mkdir -p $(VMDKOPS_BIN) $(VMDKOPS_PY) $(VMDKOPS_PY2) $(VMDKOPS_LIB) $(VMDKOPS_LIB64) $(VMDKOPS_INITD) $(UI_TO)
```
Comment (pdhamdhere) [inline esx_service/Makefile]: UI files should not be packaged with VIB until feature is ready to be released.

**Pull request closed without merging**